lib/ansible/modules/remote_management/manageiq/manageiq_config.py

```diff
@@ -0,0 +1,213 @@
+
+# set generic worker memory threshold
+- manageiq_config:
+    name: workers
+    value:
+      worker_base:
+        queue_worker_base:
+          generic_worker:
+            memory_threshold: '900.megabytes'
+'''
+
+RETURN = '''
+name:
+    description: Name of the ManageIQ config option that is being modified.
+    type: str
+    returned: always
+original_value:
+    description: The original value of the ManageIQ config option being modified before modification.
+    type: dict
+    returned: always
+value:
+    description: The value of the ManageIQ config option being modified after modification.
+    type: dict
+    returned: always
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import json
+import time
+import collections
+import copy
+
+
+# @source https://gist.github.com/angstwad/bf22d1822c38a92ec0a9
+def dict_merge(dct, merge_dct):
+    """ Recursive dict merge. Inspired by :meth:``dict.update()``, instead of
+    updating only top-level keys, dict_merge recurses down into dicts nested
+    to an arbitrary depth, updating keys. The ``merge_dct`` is merged into
+    ``dct``.
+    :param dct: dict onto which the merge is executed
+    :param merge_dct: dct merged into dct
+    :return: None
+    """
+    for k, v in merge_dct.items():
+        if (k in dct and isinstance(dct[k], dict)
+                and isinstance(merge_dct[k], collections.Mapping)):
+            dict_merge(dct[k], merge_dct[k])
+        else:
+            dct[k] = merge_dct[k]
+
+
+def get_manageiq_config_value(module, name):
+    """ Gets the current value for the given config option.
+    :param module: AnsibleModule making the call
+    :param name: ManageIQ config option name
+    :return: dict of value of the ManageIQ config option
+    """
+    returncode, out, err = module.run_command([
+        "rails",
+        "r",
+        "puts MiqServer.my_server.get_config(:%s).config.to_json" % (name)
+    ], cwd=module.params['vmdb_path'])
+    if returncode != 0:
+        raise Exception("Error getting existing value for ':%s' config: %s" % (name, err))
+
+    return json.loads(out)
+
+
+def create_expected_value(original_value, changes):
+    """ Merges changes into original value to create a merged expected value.
+    :param original_value: dict original value to merge changes into
+    :param changes: dict of changes to merge into original_value to create the expected value
+    :return: dict of the changes merged into the original_value to create an expected value
+    """
+    expected_value = copy.deepcopy(original_value)
+    dict_merge(expected_value, changes)
+    return expected_value
+
+
+def main():
+    # define the module
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(type='str', required=True),
+            value=dict(type='dict', required=False, default={}),
+            vmdb_path=dict(type='str', required=False, default='/var/www/miq/vmdb'),
```

Comment:
Please verify if type='path' can be used here, also see https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L1938